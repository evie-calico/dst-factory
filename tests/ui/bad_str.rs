use dst_factory::make_dst_factory;

#[allow(non_camel_case_types)]
type str = String;

#[make_dst_factory]
struct BadStr {
    id: i32,
    data: str,
}

fn main() {
}
