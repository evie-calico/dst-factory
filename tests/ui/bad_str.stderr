error[E0512]: cannot transmute between types of different sizes, or dependently-sized types
 --> tests/ui/bad_str.rs:6:1
  |
6 | #[make_dst_factory]
  | ^^^^^^^^^^^^^^^^^^^
  |
  = note: source type: `(*const MaybeUninit<(i32, u8)>, usize)` (128 bits)
  = note: target type: `*const BadStr` (64 bits)
  = note: this error originates in the attribute macro `make_dst_factory` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0512]: cannot transmute between types of different sizes, or dependently-sized types
 --> tests/ui/bad_str.rs:6:1
  |
6 | #[make_dst_factory]
  | ^^^^^^^^^^^^^^^^^^^
  |
  = note: source type: `(*mut (), usize)` (128 bits)
  = note: target type: `*mut BadStr` (64 bits)
  = note: this error originates in the attribute macro `make_dst_factory` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0512]: cannot transmute between types of different sizes, or dependently-sized types
 --> tests/ui/bad_str.rs:6:1
  |
6 | #[make_dst_factory]
  | ^^^^^^^^^^^^^^^^^^^
  |
  = note: source type: `(*mut u8, usize)` (128 bits)
  = note: target type: `*mut BadStr` (64 bits)
  = note: this error originates in the attribute macro `make_dst_factory` (in Nightly builds, run with -Z macro-backtrace for more info)
